k2/addons/k2-metrics/metrics/metrics.py

```diff
@@ -159,16 +159,19 @@ def __getitem__(self, name):
         return self.__getattr__(name)
 
     def __iter__(self):
-        if self.is_leaf and self._metrics is not None:
-            if self._is_single_value():
-                yield self._metrics
+        try:
+            if self.is_leaf and self._metrics is not None:
+                if self._is_single_value():
+                    yield self._metrics
+                else:
+                    yield from self._metrics
+                raise StopIteration()
             else:
-                yield from self._metrics
-            raise StopIteration()
-        else:
-            for metric in self._children.values():
-                if isinstance(metric, MetricsNamespace):
-                    yield metric
+                for metric in self._children.values():
+                    if isinstance(metric, MetricsNamespace):
+                        yield metric
+        except StopIteration:
+            return
```

Comment:
This is actually a fix for python 3.8, where `StopIteration` started causing runtime exception.